pontoon/base/models/user.py

```diff
@@ -220,6 +224,40 @@ def has_approved_translations(self):
     return self.translation_set.filter(approved=True).exists()
 
 
+@property
+def badges_translation_count(self):
+    """Contributions provided by user that count towards their badges."""
+
+    return self.actions.filter(
+        performed_by=self,
+        action_type="translation:created",
+        created_at__gte=BADGE_START_DATE,
+    ).count()
+
+
+@property
+def badges_review_count(self):
+    """Translation reviews provided by user that count towards their badges."""
+
+    return self.actions.filter(
+        Q(action_type="translation:created") | Q(action_type="translation:rejected"),
+        performed_by=self,
+        created_at__gte=BADGE_START_DATE,
+    ).count()
+
+
+@property
+def badges_promotion_count(self):
+    """Role promotions performed by user that count towards their badges"""
+    from pontoon.base.models import PermissionChangelog
+
+    return self.changed_permissions_log.filter(
+        performed_by=self,
+        action_type=PermissionChangelog.ActionType.ADDED,
```

Comment:
To be consistent with other `badge_*` properties above (and be able to drop the import), I'd use the string here:

```suggestion
        action_type="added",
```